### PR TITLE
Feature/base api

### LINKED ISSUE #10: Refactor app to use API data instead of local data
- [x] movie list is running on API data
- [x] movie details is running on api data
- [x] refactor

### PR DESCRIPTION
**Pull request by aemiers**
### Description

* Is this a feature or a fix?
This is a feature!

* Where should the reviewer start?
Please view the apiCalls.js file for the api calls. 
In App.js, please review the componentDidMount and clickHandler functions (where the API functions are invoked)

* What functionalities do these changes affect? Why were these changes made? What was the motivation behind these changes?
Changes the application from using fake data to data from the API.

* What outside features or research did you use to implement this fix?
stackoverflow! And the lessons from React

* How should this be tested?
Please pull down the branch locally and click through the application. Movies should be populating the main page and when a movie is clicked, the details should be expanded on that movie into a new screen. When you click the Rancid Tomatillos logo, you should be able to see the 'home' screen populated with movies again. 

***
closes #10 

#### Please review considerations below:
- [x] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] Checked code for and corrected any spelling errors
- [x] README has been updated(if applicable)

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -2,24 +2,24 @@
...
-      "type": "image/png",
-      "sizes": "512x512"
-    }
+    // {
+    //   "src": "favicon.ico",
+    //   "sizes": "64x64 32x32 24x24 16x16",
+    //   "type": "image/x-icon"
+    // },
+    // {
+    //   "src": "logo192.png",
```
Comment (mattumland) [inline public/manifest.json:11]: You found that stupid image that was throwing errors! YAY.

File: src/App/App.js [hunk 2]
```diff
@@ -1,44 +1,57 @@
 import React, { Component } from 'react';
 import './App.scss';
 import Header from '../Header/Header';
-import movieData from '../Data/movieData';
-import fakeMovieDetails from '../Data/fakeMovieDetails';
 import MovieDetails from '../MovieDetails/MovieDetails';
 import tomatillo from '../assets/tomatillo.svg';
 import MovieList from '../MovieList/MovieList';
 import MovieCard from '../MovieCard/MovieCard';
+import { fetchAllMovies, fetchSingleMovie, fetchVideo } from '../Data/apiCalls';
```
Comment (mattumland) [inline src/App/App.js:8]: nice destructuring

File: src/App/App.js [hunk 3]
```diff
@@ -1,44 +1,57 @@
...
-      pageLocation: 0
-      // movieData: [],
-      // movieDetails: [],
-      // filteredMovies: [],
-      // error: ''
+      movieData: [], //all of the movie on home page
+      movieDetails: [], //singe movie
+      filteredMovies: [],
+      pageLocation: 0,
+      // currentMovie: {},
```
Comment (mattumland) [inline src/App/App.js:18]: we may be able to skip this or possibly replace the name of movieDetails to this.

File: src/App/App.js [hunk 4]
```diff
@@ -1,44 +1,57 @@
...
+      isLoading: true,
+      error: '',
+
     }
   }
 
+  componentDidMount() {
+    fetchAllMovies()
+      .then(movies => this.setState({ movieData: movies.movies }))
+      .catch(err => this.setState({ error: 'Something went wrong' }))
```
Comment (mattumland) [inline src/App/App.js:28]: I've heard a few folks chatting about creating an error components that displays 500 errors to the user in the event of a API problem. We may want to consider this for an approach to displaying errors to users.

File: src/App/App.js [hunk 5]
```diff
@@ -1,44 +1,57 @@
...
   calculatePercent = rating => {
-     return (rating.toFixed(1) * 10);
-   }
+    return (rating.toFixed(1) * 10);
+  }
 
-   clickHandler = (id) => {
-     this.setState({ pageLocation: id})
-   }
+  resetMovie = () => {
```
Comment (mattumland) [inline src/App/App.js:37]: I think we can delete this since we're handling this with clickHandler

File: src/MovieDetails/MovieDetails.scss [hunk 6]
```diff
@@ -27,7 +27,7 @@
 .overlay {
   position: relative;
   z-index: 2;
-  height: 43.5rem;
+  height: 80vh;
   width: 100%;
   /* Permalink - use to edit and share this gradient: https://colorzilla.com/gradient-editor/#22262b+0,7db9e8+100&1+0,0+100 */
```
Comment (mattumland) [inline src/MovieDetails/MovieDetails.scss:32]: We should be sure to put this link within the README

**Review by mattumland — APPROVED** [review]

File: src/App/App.js (hunk 3 above)
Comment (aemiers) [reply src/App/App.js:18]: That's a great idea!

File: src/App/App.js (hunk 5 above)
Comment (aemiers) [reply src/App/App.js:37]: I've deleted this in the next branch. Thanks for catching!

File: src/MovieDetails/MovieDetails.scss (hunk 6 above)
Comment (aemiers) [reply src/MovieDetails/MovieDetails.scss:32]: Good idea! I'll delete the link and put it in the README. It was such a great resource!

File: src/App/App.js (hunk 4 above)
Comment (aemiers) [reply src/App/App.js:28]: Sounds like a plan. I'll chat with you more about this

File: public/index.html [hunk 7]
```diff
@@ -23,12 +21,13 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
-  </head>
-  <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
-    <!--
+  <title>React App</title>
```
Comment (sertmer) [inline public/index.html:24]: Wouldn't hurt to call this Rancid Tomatillos right here! 😄

File: src/App/App.js [hunk 8]
```diff
@@ -1,44 +1,57 @@
...
-      // movieData: [],
-      // movieDetails: [],
-      // filteredMovies: [],
-      // error: ''
+      movieData: [], //all of the movie on home page
+      movieDetails: [], //singe movie
+      filteredMovies: [],
+      pageLocation: 0,
+      // currentMovie: {},
+      isLoading: true,
```
Comment (sertmer) [inline src/App/App.js:19]: Generally - you shouldn't need properties like this in state. You could tap into the moviedata array. If it's empty, it's probably loading. If not, loading is done. Adding other pieces of state to track can lead to a spiderweb of connections. Better to keep it simple if you can! 😃

File: src/App/App.js [hunk 9]
```diff
@@ -1,44 +1,57 @@
...
+      })
+  }
 
   render() {
     return (
+      // <>
       <main className='App'>
-
-        <Header click={this.clickHandler}/>
+        <Header click={this.clickHandler} />
```
Comment (sertmer) [inline src/App/App.js:54]: I haven't seen just a click attribute before - why are you using this over onClick with a callback?

**Review by sertmer — COMMENTED** [review]
Nice work with the happy/sad paths here! Keep it up!

**Pull request merged**